app/app.go

```diff
@@ -467,6 +522,48 @@ func New(
 
 	planName := "v4.0.0"
 	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// set the ICS27 consensus version so InitGenesis is not run
+		fromVM[icatypes.ModuleName] = icaModule.ConsensusVersion()
```

Comment:
Because we're calling `InitModule` later in upgrade handler. 

Ref:

- https://github.com/cosmos/ibc-go/blob/46e020640e66f9043c14c53a4d215a5b457d6703/modules/apps/27-interchain-accounts/module.go#L107
- https://github.com/cosmos/ibc-go/blob/v3.0.0/docs/migrations/v2-to-v3.md#upgrade-proposal